wetterdienst/core/scalar/request.py

```diff
@@ -529,67 +571,99 @@ def nearby_number(
+        df[Columns.DISTANCE.value] = distances_km
 
-        if metadata_location.empty:
+        if df.empty:
             log.warning(
                 f"No weather stations were found for coordinate "
-                f"{latitude}°N and {longitude}°E "
+                f"{latitude}°N and {longitude}°E and number {number}"
             )
 
-        # TODO: use StationsResult here
-        # TODO: add filter for number filtering
-
-        result = StationsResult(self, metadata_location.reset_index(drop=True))
+        result = StationsResult(self, df.reset_index(drop=True))
 
         return result
 
-    def nearby_radius(
-        self,
-        latitude: float,
-        longitude: float,
-        max_distance_in_km: int,
-    ) -> "StationsResult":
+    def nearby_distance(
+        self, latitude: float, longitude: float, distance: float, unit: str = "km"
+    ) -> StationsResult:
         """
         Wrapper for get_nearby_stations_by_distance using the given parameter set.
         Returns nearest stations defined by distance (km).
 
         :param latitude: latitude in degrees
         :param longitude: longitude in degrees
-        :param max_distance_in_km: distance (km) for which stations will be selected
+        :param distance: distance (km) for which stations will be selected
+        :param unit: unit string for conversion
         :return: pandas.DataFrame with station information for the selected stations
         """
+        distance = float(distance)
+
         # Theoretically a distance of 0 km is possible
-        if max_distance_in_km < 0:
-            raise ValueError("'max_distance_in_km' has to be at least 0.0.")
+        if distance < 0:
+            raise ValueError("'distance' has to be at least 0.0")
+
+        unit = unit.strip()
 
-        metadata = self.all().df
+        distance_in_km = guess(distance, unit, [Distance]).km
 
+        # TODO: replace the repeating call to self.all()
         all_nearby_stations = self.nearby_number(
-            latitude, longitude, metadata.shape[0]
+            latitude, longitude, self.all().df.shape[0]
         ).df
 
-        nearby_stations_in_distance = all_nearby_stations[
-            all_nearby_stations[Columns.DISTANCE_TO_LOCATION.value]
-            <= max_distance_in_km
+        df = all_nearby_stations[
+            all_nearby_stations[Columns.DISTANCE.value] <= distance_in_km
         ]
 
-        # TODO: use StationsResult here
-        # TODO: add filter for radius filtering
+        if df.empty:
+            log.warning(
+                f"No weather stations were found for coordinate "
+                f"{latitude}°N and {longitude}°E and distance {distance_in_km}km"
+            )
 
-        result = StationsResult(
-            self, nearby_stations_in_distance.reset_index(drop=True)
-        )
+        result = StationsResult(stations=self, df=df.reset_index(drop=True))
+
+        return result
+
+    def bbox(
```

Comment:
Consequently, this new method might also be named `filter_by_bbox()`. However, I like both the long and short variants. Maybe let's implement all `filter_*()` methods first and then add aliases later?

Reply:
I thought the way it is right now is fine, because the two filter methods really only do filtering based on the table while those others are applying more like a geo filter. However the moment I'm writing this, I'm agreeing with you... Filter_by_bbox, filter_by_number/rank, filter_by distance would fit well